api/src/main/java/jakarta/data/metamodel/constraint/LikePatternRecord.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel.constraint;
+
+import jakarta.data.metamodel.restrict.Operator;
+
+import java.util.Objects;
+
+record LikePatternRecord(String pattern, boolean caseSensitive)
+        implements Like {
+
+    public static final char CHAR_WILDCARD = '_';
+    public static final char STRING_WILDCARD = '%';
+    public static final char ESCAPE = '\\';
+
+    public LikePatternRecord {
+        Objects.requireNonNull(pattern, "Pattern must not be null");
+    }
+
+    public LikePatternRecord(String pattern) {
+        this(pattern, true);
+    }
+
+    public LikePatternRecord(String pattern, char charWildcard, char stringWildcard) {
+        this(translate(pattern, charWildcard, stringWildcard));
+    }
+
+    @Override
+    public LikePatternRecord ignoreCase() {
+        return new LikePatternRecord(pattern, false);
+    }
+
+    @Override
+    public Operator operator() {
+        return Operator.LIKE;
+    }
+
+    @Override
+    public String toString() {
+        return "'" + pattern + "'";
```

Comment:
Even after this update, it will look the same as the output of LikeLiteralRecord, so we need to do something further to distinguish. Possibly adding in ESCAPED=true/false?  I can't add it here because that information might have disappeared, so we need to look into that as well.
```suggestion
        return "LIKE '" + pattern + "'" + (caseSensitive ? "" : " IGNORE CASE");
```